tests/src/test/java/alluxio/client/rest/S3ClientRestApiTest.java

```diff
@@ -1302,11 +1295,25 @@ public void listParts() throws Exception {
       ListPartsResult.Part part = parts.get(partNumber - 1);
       Assert.assertEquals(partNumber, part.getPartNumber());
       URIStatus status = mFileSystem.getStatus(
-          new AlluxioURI(tmpDir + AlluxioURI.SEPARATOR + Integer.toString(partNumber)));
+          new AlluxioURI(tmpDir + AlluxioURI.SEPARATOR + partNumber));
       Assert.assertEquals(S3RestUtils.toS3Date(status.getLastModificationTimeMs()),
           part.getLastModified());
       Assert.assertEquals(status.getLength(), part.getSize());
     }
+
+    // TODO(czhu): get this 404 to manifest
+    /*
+    // Call ListParts as a separate FileSystem user
+    // Verify 404 HTTP status & NoSuchBucket S3 error code
+    HttpURLConnection connection = new TestCase(mHostname, mPort, mBaseUri,
+        objectKey, ImmutableMap.of("uploadId", uploadId), HttpMethod.GET,
+        TestCaseOptions.defaults().setAuthorization("AWS4-HMAC-SHA256 Credential=dummy/20220830"))
+        .execute();
+    Assert.assertEquals(404, connection.getResponseCode());
+    S3Error response =
+        new XmlMapper().readerFor(S3Error.class).readValue(connection.getErrorStream());
+    Assert.assertEquals(S3ErrorCode.Name.NO_SUCH_BUCKET, response.getCode());
+    */
```

Comment:
For some reason, I can't seem to limit the scope of files seen by this `"dummy"` user. The `listStatus` gRPC call shows both files even though the directory is not owned by this user. I need to double check that the group is not shared, and that the permission bits I set actually were present.